src/cryptonote_core/loki_name_system.cpp

```diff
@@ -810,31 +810,34 @@ bool validate_lns_name(mapping_type type, std::string name, std::string *reason)
     if (check_condition(name == "localhost.loki", reason, "LNS type=", type, ", specifies mapping from name->value using protocol reserved name=", name))
       return false;
 
-    // Must start with alphanumeric
-    if (check_condition(!char_is_alphanum(name.front()), reason, "LNS type=", type, ", specifies mapping from name->value where the name does not start with an alphanumeric character, name=", name))
-      return false;
-
-    char const SHORTEST_DOMAIN[] = "a.loki";
-    if (check_condition((name.size() < static_cast<int>(loki::char_count(SHORTEST_DOMAIN))), reason, "LNS type=", type, ", specifies mapping from name->value where the name is shorter than the shortest possible name=", SHORTEST_DOMAIN, ", given name=", name))
+    auto constexpr SHORTEST_DOMAIN = "a.loki"sv;
+    if (check_condition(name.size() < SHORTEST_DOMAIN.size(), reason, "LNS type=", type, ", specifies mapping from name->value where the name is shorter than the shortest possible name=", SHORTEST_DOMAIN, ", given name=", name))
       return false;
 
     // Must end with .loki
-    char const SUFFIX[]     = ".loki";
-    char const *name_suffix = name.data() + (name.size() - loki::char_count(SUFFIX));
-    if (check_condition((memcmp(name_suffix, SUFFIX, loki::char_count(SUFFIX)) != 0), reason, "LNS type=", type, ", specifies mapping from name->value where the name does not end with the domain .loki, name=", name))
+    auto constexpr SUFFIX = ".loki"sv;
+    if (check_condition(name_view.substr(name_view.size() - SUFFIX.size()) == SUFFIX, reason, "LNS type=", type, ", specifies mapping from name->value where the name does not end with the domain .loki, name=", name))
```

Comment:
This needs to be `name_view.substr(...) != SUFFIX`

Reply:
That code was kind of gross and so I added a `tools::ends_with` in a later commit in here (apparently I missed backporting the change into this commit); the current code is:

```C++
if (check_condition(!tools::ends_with(name_view, SUFFIX), ...
```

Reply:
(C++20 string_view has an `ends_with` but it might be just a *touch* to early to switch to C++20.  Even though it has https://en.cppreference.com/w/cpp/utility/source_location which I really, really want.)